rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,284 @@
+- Remove management section `uiExport` (required for migration)
+- Simple API that is designed in keeping with new platform principles
+  - This includes being rendering-framework-agnostic... You should be able to build your management section UI however you'd like
+- Clear separation of app/UI code and service code, even if both live within the same plugin
+- Flexibility to potentially support alternate layouts in the future (see mockups in [reference section](#reference) below)
+
+# Basic example
+This API is influenced heavily by the [application service mounting RFC](https://github.com/elastic/kibana/blob/master/rfcs/text/0004_application_service_mounting.md). The intent is to make the experience consistent with that service; the Management section is basically one big app with a bunch of registered "subapps".
+
+```ts
+// my_plugin/public/plugin.ts
+
+export class MyPlugin {
+  setup(core, { management }) {
+    management.sections.register({
+      id: 'my-section',
+      title: 'My Main Section', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 10,
+      euiIconType: 'iconName',
+    });
+    management.sections.registerLink('my-section', {
+      id: 'my-link',
+      title: 'My Link', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 20,
+      async mount(context, params) {
+        const { renderApp } = await import('./my-section');
+        return renderApp(context, params);
+      }
+    });
+  }
+}
+ 
+// my_plugin/public/my-section.tsx
+
+export function renderApp(context, { basePath, element }) {
+  ReactDOM.render(
+    // `basePath` would be `/app/management/my-section/my-link`
+    <MyApp basename={basePath} />,
+    element
+  );
+ 
+  // return value must be a function that unmounts (just like Core Application Service)
+  return () => ReactDOM.unmountComponentAtNode(element);
+}
+```
+ 
+# Detailed design
+
+```ts
+interface ManagementSetup {
+  sections: SectionsService;
+}
+ 
+interface SectionsService {
+  register: Register;
+  registerLink: RegisterLink;
+  getAvailable: () => Section[]; // filtered based on capabilities
+  get: (id: string) => Section | undefined;
+  getLink: (id: string) => Link | undefined;
+}
+ 
+type Register = ({
+  id: string;
+  title: string;
+  description: string; // not used in current UI, but possibly in future
+  order?: number;
+  euiIconType?: string; // takes precedence over `icon` property.
+  icon?: string; // URL to image file; fallback if no `euiIconType`
+}) => Section | Error;
+ 
+type RegisterLink = (sectionId: string, {
+  id: string;
+  title: string;
+  description: string; // not used in current UI, but possibly in future
+  order?: number;
+  mount: ManagementSectionMount;
+}) => Link | Error;
+ 
+type Unmount = () => Promise<void> | void;
+ 
+type ManagementSectionMount = (
+  context: AppMountContext, // provided by core.ApplicationService
```

Comment:
Perhaps this is not entirely true from a technical point of view. Contexts have different responsibilities and (in theory) different interfaces. 
Anyway it's okay for a while.